App/Redux/PhotoViewingRedux.ts

```diff
@@ -4,21 +4,22 @@ import { ThreadId, Photo, PhotoId, Thread, ThreadName, Mnemonic } from '../Model
 
 const actions = {
   insertThread: createAction('INSERT_THREAD', (resolve) => {
-    return (thread: Thread) => resolve({ thread })
+    return (id: ThreadId, name: string) => resolve({ id, name })
   }),
   addThreadRequest: createAction('ADD_THREAD_REQUEST', (resolve) => {
-    return (name: ThreadName) => resolve({ name })
+    return (name: string, options?: { navigate?: boolean, sharePhoto?: { imageId: PhotoId, comment?: string } }) => resolve({ name }, options)
```

Comment:
The `addThreadRequest` now accepts a metadata object where you can optionally specify that you want to navigate to or share a photo to the thread after it is created.